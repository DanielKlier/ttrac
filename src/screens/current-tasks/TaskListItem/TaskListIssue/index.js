import * as React from 'react';
import PropTypes from 'prop-types';
import './Styles.css';

const TaskListIssue = props => (
    <span className="task-list-issue">
      <a href="/">{props.issue}</a>
    </span>
);

TaskListIssue.propTypes = {
    issue: PropTypes.string
};

TaskListIssue.defaultProps = {
    issue: ''
};

export default TaskListIssue;
