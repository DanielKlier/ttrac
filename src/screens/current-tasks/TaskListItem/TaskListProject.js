import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {findDOMNode} from 'react-dom';
import {noop} from 'lodash';
import {Button, Glyphicon, Overlay, Popover} from 'react-bootstrap';
import FindProjectDialog from '../../../components/FindProjectDialog/index';
import uuid from 'uuid';

class TaskListProject extends Component {
    constructor() {
        super();

        this.state = {
            isOpen: false
        };

        this.id = 'FindProjectDialog_' + uuid();

        this.openPopoverTarget = null;
    }

    render() {
        return (
            <div>
                <Button ref={ref => this.openPopoverTarget = ref} onClick={() => this.toggleOpen()}>
                    {this.renderProject()}
                </Button>
                <Overlay target={props => findDOMNode(this.openPopoverTarget)}
                         show={this.state.isOpen} placement="bottom" rootClose
                         onHide={() => this.toggleOpen(false)}>
                    <Popover id={this.id}>
                        <FindProjectDialog onSelectProject={this.props.onProjectChanged}/>
                    </Popover>
                </Overlay>
            </div>
        );
    }

    renderProject() {
        if (this.props.project) {
            return (
                <span>{this.props.project.title}</span>
            );
        }
        else {
            return (
                <Glyphicon glyph="folder-close"/>
            );
        }
    }

    toggleOpen = (isOpen) => {
        // Toggle semantics
        if (isOpen === undefined) {
            isOpen = !this.state.isOpen;
        }

        this.setState({isOpen});
    };
}

TaskListProject.propTypes = {
    project: PropTypes.object,
    onProjectChanged: PropTypes.func
};

TaskListProject.defaultProps = {
    project: null,
    onProjectChanged: noop
};

export default TaskListProject;
