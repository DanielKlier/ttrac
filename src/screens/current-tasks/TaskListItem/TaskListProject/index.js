import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {findDOMNode} from 'react-dom';
import {noop} from 'lodash';
import {Button, Glyphicon, Label, Overlay, Popover} from 'react-bootstrap';
import FindProjectDialog from '../../../../components/FindProjectDialog/index';
import uuid from 'uuid';
import './Styles.css';

class TaskListProject extends Component {
    constructor() {
        super();

        this.state = {
            isOpen: false
        };

        this.id = 'FindProjectDialog_' + uuid();

        this.openPopoverTarget = null;
    }

    render() {
        return (
            <div className="task-list-project">
                <div ref={ref => this.openPopoverTarget = ref} onClick={() => this.toggleOpen()}>
                    {this.renderProject()}
                </div>
                <Overlay target={props => findDOMNode(this.openPopoverTarget)}
                         show={this.state.isOpen} placement="bottom"
                         onHide={() => this.toggleOpen(false)}>
                    <Popover id={this.id}>
                        <FindProjectDialog onSelectProject={this.onProjectChanged}/>
                    </Popover>
                </Overlay>
            </div>
        );
    }

    renderProject() {
        if (this.props.project) {
            return (
                <div className="task-list-project-inner">
                    <Label style={{backgroundColor: this.props.project.color}}>
                        {this.props.project.code}
                    </Label>
                    <span className="project-title">{this.props.project.title}</span>
                </div>
            );
        }
        else {
            return (
                <div className="add-project-button"><Button><Glyphicon glyph="folder-close"/></Button></div>
            );
        }
    }

    toggleOpen = (isOpen) => {
        // Toggle semantics
        if (isOpen === undefined) {
            isOpen = !this.state.isOpen;
        }

        this.setState({isOpen});
    };

    onProjectChanged = (project) => {
        this.toggleOpen(false);
        this.props.onProjectChanged(project);
    };
}

TaskListProject.propTypes = {
    project: PropTypes.object,
    onProjectChanged: PropTypes.func
};

TaskListProject.defaultProps = {
    project: null,
    onProjectChanged: noop
};

export default TaskListProject;
