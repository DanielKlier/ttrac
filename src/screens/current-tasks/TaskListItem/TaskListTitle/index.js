import * as React from 'react';
import PropTypes from 'prop-types';
import {noop} from 'lodash';
import ToggleDisplay from '../../../../components/ToggleDisplay';
import './Styles.css';

export default class TaskListTitle extends React.Component {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        title: PropTypes.string,
        onTextChanged: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        title: '',
        onTextChanged: noop
    };

    constructor(props) {
        super(props);

        this.state = {
            isEditing: !props.title,
            text: this.props.title || ''
        };

        this.titleInput = null;
    }

    render() {

        return (
            <h4 className="task-list-description">
                <ToggleDisplay show={!this.state.isEditing && this.state.text}>
                    <span className="title" onClick={() => this.showEditInput()}>
                        {this.state.text}
                    </span>
                </ToggleDisplay>
                <ToggleDisplay show={this.state.isEditing || !this.state.text}>
                    <input type="text"
                           className="title-input"
                           ref={el => {
                               this.titleInput = el;
                           }}
                           value={this.state.text}
                           onBlur={(e) => this.stopEditing(e.target.value)}
                           placeholder="Click to add a title"
                           onChange={
                               (e) => this.textInputChanged(e.target.value)
                           }
                    />
                </ToggleDisplay>
            </h4>
        );
    }

    showEditInput() {
        const input = this.titleInput;
        this.setState({isEditing: true}, () => {
            input.focus();
            input.select();
        });
    }

    stopEditing() {
        this.setState({isEditing: false}, () => {
            this.props.onTextChanged(this.state.text);
        });
    }

    textInputChanged(value) {
        this.setState({text: value});
    }
}
