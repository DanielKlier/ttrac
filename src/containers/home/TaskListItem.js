import * as React from 'react';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListShortDescription from './TaskListShortDescription';
import {Button, Glyphicon} from 'react-bootstrap';

export default props => (
    <li className="list-group-item task-list-item">
        <div className="task-list-item-badges">
            <TaskListHours hours={props.hours} className="pull-left"/>
            {props.issue && <TaskListIssue issue={props.issue}/>}
        </div>
        <div className="task-list-item-main">
            <TaskListShortDescription description={props.description}/>
        </div>
        <div className="task-list-item-buttons">
            <Button>
                <Glyphicon glyph="time"/>
            </Button>
        </div>
    </li>
);