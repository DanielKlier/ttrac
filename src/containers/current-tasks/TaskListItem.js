import * as React from 'react';
import PropTypes from 'prop-types';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import {Button, Glyphicon} from 'react-bootstrap';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static PropTypes = {
        elapsedTime           : PropTypes.number,
        jiraIssue             : PropTypes.string,
        title                 : PropTypes.string,
        taskIsRunning         : PropTypes.bool,
        runningTask           : PropTypes.object,
        onTaskTitleChanged    : PropTypes.func.isRequired,
        onStartProgressClicked: PropTypes.func.isRequired,
        onStopProgressClicked : PropTypes.func.isRequired,
        onDeleteTaskClicked   : PropTypes.func.isRequired
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <Button onClick={() => props.onDeleteTaskClicked()}>
                        <Glyphicon glyph="trash"/>
                    </Button>
                    {props.taskIsRunning === false &&
                    <Button onClick={() => props.onStartProgressClicked()}>
                        <Glyphicon glyph="time"/>
                    </Button>
                    }
                    {props.taskIsRunning === true &&
                    <Button onClick={() => props.onStopProgressClicked()}>
                        <Glyphicon glyph="pause"/>
                    </Button>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
