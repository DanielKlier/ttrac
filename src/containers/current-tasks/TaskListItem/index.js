import * as React from 'react';
import PropTypes from 'prop-types';
import {identity} from 'lodash';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';
import StartProgressButton from './StartProgressButton';
import StopProgressButton from './StopProgressButton';
import DeleteButton from '../../../components/DeleteButton';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        elapsedTime: PropTypes.number,
        jiraIssue: PropTypes.string,
        title: PropTypes.string,
        taskIsRunning: PropTypes.bool,
        runningTask: PropTypes.object,
        onTaskTitleChanged: PropTypes.func,
        onStartProgressClicked: PropTypes.func,
        onStopProgressClicked: PropTypes.func,
        onDeleteTaskClicked: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        elapsedTime: 0,
        jiraIssue: '',
        title: '',
        taskIsRunning: false,
        runningTask: null,
        onTaskTitleChanged: identity,
        onStartProgressClicked: identity,
        onStopProgressClicked: identity,
        onDeleteTaskClicked: identity
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <DeleteButton onClick={() => props.onDeleteTaskClicked()}/>
                    {props.taskIsRunning === false &&
                    <StartProgressButton onClick={() => props.onStartProgressClicked()}/>
                    }
                    {props.taskIsRunning === true &&
                    <StopProgressButton onClick={() => props.onStopProgressClicked()}/>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
