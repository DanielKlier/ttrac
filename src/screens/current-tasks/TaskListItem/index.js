import * as React from 'react';
import PropTypes from 'prop-types';
import {noop} from 'lodash';
import TaskListHours from './TaskListHours';
import TaskListIssue from './TaskListIssue';
import TaskListTitle from './TaskListTitle';
import TimeCounter from './TimeCounter';
import StartProgressButton from './StartProgressButton';
import StopProgressButton from './StopProgressButton';
import DeleteButton from '../../../components/DeleteButton';
import TaskListProject from './TaskListProject';

class TaskListItem extends React.PureComponent {
    // noinspection JSUnusedGlobalSymbols
    static propTypes = {
        elapsedTime: PropTypes.number,
        jiraIssue: PropTypes.string,
        title: PropTypes.string,
        taskIsRunning: PropTypes.bool,
        runningTask: PropTypes.object,
        onTaskTitleChanged: PropTypes.func,
        onStartProgressClicked: PropTypes.func,
        onStopProgressClicked: PropTypes.func,
        onDeleteTaskClicked: PropTypes.func,
        project: PropTypes.object,
        onProjectChanged: PropTypes.func
    };

    // noinspection JSUnusedGlobalSymbols
    static defaultProps = {
        elapsedTime: 0,
        jiraIssue: '',
        title: '',
        taskIsRunning: false,
        runningTask: null,
        onTaskTitleChanged: noop,
        onStartProgressClicked: noop,
        onStopProgressClicked: noop,
        onDeleteTaskClicked: noop,
        project: null,
        onProjectChanged: noop
    };

    render() {
        const props      = this.props;
        const classNames = [
            'list-group-item',
            'task-list-item'
        ];
        if (props.taskIsRunning) {
            classNames.push('running');
        }
        const className = classNames.join(' ');

        return (
            <li className={className}>
                <div className="task-list-item-badges">
                    {props.taskIsRunning &&
                    <TimeCounter startDate={props.runningTask.startDate}/>
                    }
                    {!props.taskIsRunning &&
                    <TaskListHours elapsedTime={props.elapsedTime}
                                   className="pull-left"/>
                    }
                    {props.jiraIssue &&
                    <TaskListIssue jiraIssue={props.jiraIssue}/>}
                </div>
                <div className="task-list-item-project">
                    <TaskListProject project={props.project}
                                     onProjectChanged={props.onProjectChanged}/>
                </div>
                <div className="task-list-item-main">
                    <TaskListTitle title={props.title}
                                   onTextChanged={
                                       text => props.onTaskTitleChanged(text)
                                   }/>
                </div>
                <div className="task-list-item-buttons">
                    <DeleteButton onClick={() => props.onDeleteTaskClicked()}/>
                    {props.taskIsRunning === false &&
                    <StartProgressButton onClick={() => props.onStartProgressClicked()}/>
                    }
                    {props.taskIsRunning === true &&
                    <StopProgressButton onClick={() => props.onStopProgressClicked()}/>
                    }
                </div>
            </li>
        );
    }
}

export default TaskListItem;
