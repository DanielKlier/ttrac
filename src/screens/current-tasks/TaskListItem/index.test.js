import React from 'react';
import TaskListItem from './index';
import {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';

Date.now = jest.fn(() => new Date('2017-03-09T19:33:00'));

test('TaskListItem renders correctly', () => {
    let component = mount(
        <TaskListItem elapsedTime={3600000} jiraIssue="PRJ-123"
                      title="My Task"/>
    );
    let tree      = toJSON(component);

    expect(tree).toMatchSnapshot();

    component = mount(
        <TaskListItem elapsedTime={3600000} jiraIssue="PRJ-123" title="My Task"
                      taskIsRunning={true}
                      runningTask={{
                          startDate: new Date('2017-03-09T18:00:00')
                      }}/>
    );
    tree      = toJSON(component);

    expect(tree).toMatchSnapshot();
});

test('TaskListItem calls the handler when the delete button is clicked', () => {

    const handler   = jest.fn();
    const component = mount(<TaskListItem onDeleteTaskClicked={handler}/>);

    component.find('DeleteButton').simulate('click');

    expect(handler).toHaveBeenCalled();
});

test('TaskListItem calls the handler if the start progress button is clicked', () => {

    const handler1  = jest.fn();
    const handler2  = jest.fn();
    const component = mount(<TaskListItem onStartProgressClicked={handler1}
                                          onStopProgressClicked={handler2}
                                          taskIsRunning={false}
                                          runningTask={{startDate: new Date('2017-03-09T18:00:00')}}
    />);

    component.find('StartProgressButton').simulate('click');
    expect(handler1).toHaveBeenCalled();

    component.setProps({taskIsRunning: true});
    component.update();

    component.find('StopProgressButton').simulate('click');
    expect(handler2).toHaveBeenCalled();
});

test('TaskListItem calls the title changed handler', () => {

    const handler = jest.fn();

    const component = mount(<TaskListItem title={''} onTaskTitleChanged={handler}/>);

    // Input of text and then blur should trigger a change event
    component.find('.title-input').simulate('change', {target: {value: 'hello'}});
    component.find('.title-input').simulate('blur');

    expect(handler).toBeCalledWith('hello');
});
