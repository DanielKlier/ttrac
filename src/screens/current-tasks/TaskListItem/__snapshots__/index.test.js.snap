// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem renders correctly 1`] = `
<TaskListItem
  elapsedTime={3600000}
  jiraIssue="PRJ-123"
  onDeleteTaskClicked={[Function]}
  onStartProgressClicked={[Function]}
  onStopProgressClicked={[Function]}
  onTaskTitleChanged={[Function]}
  runningTask={null}
  taskIsRunning={false}
  title="My Task"
>
  <li
    className="list-group-item task-list-item"
  >
    <div
      className="task-list-item-badges"
    >
      <TaskListHours
        className="pull-left"
        elapsedTime={3600000}
      >
        <span
          className="task-list-hours"
        >
          <Label
            bsClass="label"
            bsStyle="info"
          >
            <span
              className="label label-info"
            >
              60m 0s
            </span>
          </Label>
        </span>
      </TaskListHours>
      <TaskListIssue
        issue=""
        jiraIssue="PRJ-123"
      >
        <span
          className="task-list-issue"
        >
          <a
            href="/"
          />
        </span>
      </TaskListIssue>
    </div>
    <div
      className="task-list-item-main"
    >
      <TaskListTitle
        onTextChanged={[Function]}
        title="My Task"
      >
        <h4
          className="task-list-description"
        >
          <toggleDisplay
            show="My Task"
          >
            <span
              className=""
            >
              <span
                className="title"
                onClick={[Function]}
              >
                My Task
              </span>
            </span>
          </toggleDisplay>
          <toggleDisplay
            show={false}
          >
            <span
              className="hidden"
            >
              <input
                className="title-input"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Click to add a title"
                type="text"
                value="My Task"
              />
            </span>
          </toggleDisplay>
        </h4>
      </TaskListTitle>
    </div>
    <div
      className="task-list-item-buttons"
    >
      <DeleteButton
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="trash"
            >
              <span
                className="glyphicon glyphicon-trash"
              />
            </Glyphicon>
          </button>
        </Button>
      </DeleteButton>
      <StartProgressButton
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="time"
            >
              <span
                className="glyphicon glyphicon-time"
              />
            </Glyphicon>
          </button>
        </Button>
      </StartProgressButton>
    </div>
  </li>
</TaskListItem>
`;

exports[`TaskListItem renders correctly 2`] = `
<TaskListItem
  elapsedTime={3600000}
  jiraIssue="PRJ-123"
  onDeleteTaskClicked={[Function]}
  onStartProgressClicked={[Function]}
  onStopProgressClicked={[Function]}
  onTaskTitleChanged={[Function]}
  runningTask={
    Object {
      "startDate": 2017-03-09T18:00:00.000Z,
    }
  }
  taskIsRunning={true}
  title="My Task"
>
  <li
    className="list-group-item task-list-item running"
  >
    <div
      className="task-list-item-badges"
    >
      <TimeCounter
        startDate={2017-03-09T18:00:00.000Z}
      >
        <span
          className="task-list-hours"
        >
          <Label
            bsClass="label"
            bsStyle="warning"
          >
            <span
              className="label label-warning"
            >
              0s
            </span>
          </Label>
        </span>
      </TimeCounter>
      <TaskListIssue
        issue=""
        jiraIssue="PRJ-123"
      >
        <span
          className="task-list-issue"
        >
          <a
            href="/"
          />
        </span>
      </TaskListIssue>
    </div>
    <div
      className="task-list-item-main"
    >
      <TaskListTitle
        onTextChanged={[Function]}
        title="My Task"
      >
        <h4
          className="task-list-description"
        >
          <toggleDisplay
            show="My Task"
          >
            <span
              className=""
            >
              <span
                className="title"
                onClick={[Function]}
              >
                My Task
              </span>
            </span>
          </toggleDisplay>
          <toggleDisplay
            show={false}
          >
            <span
              className="hidden"
            >
              <input
                className="title-input"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Click to add a title"
                type="text"
                value="My Task"
              />
            </span>
          </toggleDisplay>
        </h4>
      </TaskListTitle>
    </div>
    <div
      className="task-list-item-buttons"
    >
      <DeleteButton
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="trash"
            >
              <span
                className="glyphicon glyphicon-trash"
              />
            </Glyphicon>
          </button>
        </Button>
      </DeleteButton>
      <StopProgressButton
        onClick={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <Glyphicon
              bsClass="glyphicon"
              glyph="pause"
            >
              <span
                className="glyphicon glyphicon-pause"
              />
            </Glyphicon>
          </button>
        </Button>
      </StopProgressButton>
    </div>
  </li>
</TaskListItem>
`;
